packages/integration-tests/src/entities/BookReview.ts

```diff
@@ -1,24 +1,12 @@
+import { hasOneThrough, Reference } from "joist-orm";
 import { Author, BookReviewCodegen, bookReviewConfig } from "./entities";
-import { CustomReference, Reference } from "joist-orm";
 
 export class BookReview extends BookReviewCodegen {
-  readonly author: Reference<BookReview, Author, never> = new CustomReference<
-    BookReview,
-    Author,
-    { book: "author" },
-    never
-  >(this, "author", {
-    load: async (review) => {
-      await review.load((r) => r.book.author);
-    },
-    get: (review) => review.book.get.author.get,
-    set: (review, author) => {
-      review.book.get.author.set(author);
-    },
-    isSet: (review) => {
-      return review.book.isSet() && review.book.get.author.isSet();
-    },
-  });
+  readonly author: Reference<BookReview, Author, never> = hasOneThrough(
+    this as BookReview,
+    { book: "author" } as const,
+    (review) => review.book.get.author.get,
```

Comment:
I'm pretty sure this could be:

```typescript
hasOneThrough(review => review.book.author);
```

Using the lens functionality, but currently the lens eval logic can only do `await`s but with some copy/paste, it could have a synchronous eval...

Reply:
(Updated to use the lens in the latest commits.)